### PR TITLE
Make Mbstring extension required

### LINKED ISSUE #9583: [BOOM-6001] Missing PHP function mb_split
> This issue has been migrated from the Forge. Read the [original ticket here](http://forge.prestashop.com/browse/BOOM-6001).

- _**Reporter:**_ Lilian
- _**Created at:**_ Mon, 16 Jul 2018 15:41:55 +0200

<p>Hi,</p>

<p>I just installed Prestashop. After my first clicks in the backoffice I encountered an Internal Error 500 after trying to display the categories or commands.</p>

<p>Thanks to the DEBUG mode I see the problem is the mb_split function that is missing.</p>

<p>After installing the php-mbstring package in my ubuntu distribution, the problem is solved.</p>

<p>As the Prestashop install process asked me to install some missing things, I think this is a check that is missing in the install process.</p>

<p>Regards</p>
        
* How to reproduce the issue ?

Download prestashop_1.7.4.0.zip
<br/>
Follow installation instructions to install it on Linux Mint 19 Tara
<br/>
Connect to the backoffice
<br/>
Enable Debug mode
<br/>
Go to Vendre / Catalogue / Categories or Vendre / Commandes / Commandes

### PR DESCRIPTION
**Pull request by eternoendless**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Mbstring was always required by Smarty because it uses `mb_split()` which is not polyfilled. This change makes this requirement explicit by adding it to the requirement checks during install and in composer.
| Type?         | bug fix
| Category?     | CO
| BC breaks?    | no (it was always required but not checked)
| Deprecations? | no
| Fixed ticket? | Fixes #9583
| How to test?  | Try to install this PR in an environment with the mbstring extension disabled.

Related doc PR: https://github.com/PrestaShop/docs/pull/436

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/17046)

### CODE REVIEW

File: composer.lock [hunk 1]
```diff
@@ -6290,16 +6290,16 @@
         },
         {
             "name": "symfony/polyfill-mbstring",
-            "version": "v1.12.0",
+            "version": "v1.13.1",
```
Comment (Progi1984) [inline composer.lock]: @eternoendless If it removed from the composer.json, why is it updated here ?

Comment (eternoendless) [reply composer.lock]: It's required by symfony:

```
$ composer depends "symfony/polyfill-mbstring"
symfony/symfony  v3.4.26  requires  symfony/polyfill-mbstring (~1.0)  
```

Comment (Progi1984) [reply composer.lock]: :ok_hand:

**Review by Progi1984 — DISMISSED** [review]

**Comment by Progi1984** [comment]
@eternoendless Need update the devdocs too : https://devdocs.prestashop.com/1.7/basics/installation/system-requirements/#extensions

**Comment by Progi1984** [comment]
And I have no see the link under the table => :door:

**Review by PierreRambaud — DISMISSED** [review]

**Comment by khouloudbelguith** [comment]
Hi @eternoendless,

It is ok
Install with `mbstring` enabled
image
Install with `mbstring` disabled
image

PS: there are some conflicts in this branch.

Thanks!

**Comment by eternoendless** [comment]
Branch rebased @PierreRambaud

**Comment by colinegin** [comment]
ping @PrestaShop/prestashop-core-developers what is this status of this PR ?

**Comment by PierreRambaud** [comment]
@colinegin Thanks. I rebase his pull request, we need for the CI and we will merge it :)

Edit: The PR was already QA valid. Only a rebase was needed

**Comment by colinegin** [comment]
You are amazing

**Review by Progi1984 — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by PierreRambaud** [comment]
Thanks @eternoendless

**Pull request merged**